### PR TITLE
Reset files into the same branch they were committted to

### PR DESCRIPTION
**Pull request by mtsgrd**
- regressed when switching to blame for hunk locking

### CODE REVIEW

File: crates/gitbutler-core/src/virtual_branches/virtual.rs [hunk 1]
```diff
@@ -2042,11 +2043,51 @@ pub fn reset_branch(
...
+            .find_commit(updated_head)
+            .map_err(anyhow::Error::from)?
+            .tree()
+            .map_err(anyhow::Error::from)?,
+        &repo
+            .find_commit(old_head)
+            .map_err(anyhow::Error::from)?
+            .tree()
+            .map_err(anyhow::Error::from)?,
+    )?;
```
Comment (mtsgrd) [inline crates/gitbutler-core/src/virtual_branches/virtual.rs:2061]: @Byron any chance you could help me make this a bit more trim?

Comment (Byron) [reply crates/gitbutler-core/src/virtual_branches/virtual.rs:2061]: Let me see….

Comment (Byron) [reply crates/gitbutler-core/src/virtual_branches/virtual.rs:2061]: Yes, and no. I pushed a commit that allows to use `?` in quite  straightforward way, which probably pays off quickly.

What follows *would* work if there was trait specialization, but it's a missing feature I keep falling for.

```patch
diff --git a/crates/gitbutler-core/src/virtual_branches/errors.rs b/crates/gitbutler-core/src/virtual_branches/errors.rs
index f802ae893..cf8473109 100644
--- a/crates/gitbutler-core/src/virtual_branches/errors.rs
+++ b/crates/gitbutler-core/src/virtual_branches/errors.rs
@@ -55,6 +55,17 @@ pub enum ResetBranchError {
     DefaultTargetNotSet(DefaultTargetNotSet),
     #[error(transparent)]
     Other(#[from] anyhow::Error),
+    #[error(transparent)]
+    Any(Box<dyn std::error::Error + Send + Sync>),
+}
+
+impl<E> From<E> for ResetBranchError
+where
+    E: std::error::Error + Send + Sync,
+{
+    fn from(err: E) -> Self {
+        ResetBranchError::Any(err.into())
+    }
 }
 
 impl ErrorWithContext for ResetBranchError {
diff --git a/crates/gitbutler-core/src/virtual_branches/virtual.rs b/crates/gitbutler-core/src/virtual_branches/virtual.rs
index 0caff1d5f..ccf1606a7 100644
--- a/crates/gitbutler-core/src/virtual_branches/virtual.rs
+++ b/crates/gitbutler-core/src/virtual_branches/virtual.rs
@@ -2056,16 +2056,8 @@ pub fn reset_branch(
     let repo = &project_repository.git_repository;
     let diff = trees(
         repo,
-        &repo
-            .find_commit(updated_head)
-            .map_err(anyhow::Error::from)?
-            .tree()
-            .map_err(anyhow::Error::from)?,
-        &repo
-            .find_commit(old_head)
-            .map_err(anyhow::Error::from)?
-            .tree()
-            .map_err(anyhow::Error::from)?,
+        &repo.find_commit(updated_head)?.tree()?,
+        &repo.find_commit(old_head)?.tree()?,
     )?;
 
     // Assign the new hunks to the branch we're working on.
```

Also Rust nightly doesn't allow that or have it in a working condition, unfortunately. But probably it's for the better here as what's pushed is simpler.

Comment (mtsgrd) [reply crates/gitbutler-core/src/virtual_branches/virtual.rs:2061]: Thanks! 🙏

**Pull request merged**